### ISSUE TITLE
Payment requests fail due to invalid access token....

### ISSUE THREAD

**Issue by ambrose-at-cordic** [sumup/sumup-android-sdk#74]
We have been noticing that our customers using our driver app to process chip and pin payments via the Sumup SDK are failing due to "access token invalid" errors. 

When we perform payments we always request a fresh token for each transaction yet, we have recently noticed that payments have been failing due to the token being invalid.
I have also noted that the process in obtaining a token has changed as seems the grant_type should be "authorization_code" whereas we have been using "password" in our app. 

I have also noticed that you have to make an authorization request to obatin the code and this would be used when requesting the token with the grant_type of "authorization_code".

Would this be the cause of these errors?

**Comment by JullianSU** [comment]
Hi, 

Would you be able to share the set up you have and your client_id details at **integration@sumup.com**? From there, we will be able to look at your issue. 

Many thanks,

**Comment by ambrose-at-cordic** [comment]
Hello @JullianSU , I had recently come back to looking at this as I had updated our app to use v3.2.0 of the SDK and had a recurrence of this issue. I had also shared the setup and client_id details at integration@sumup.com on thursday (20/06/2019). Just wondered if you had received it and again, if there is an issue in using the Resource Owner Password Credentials grant flow to obtain access token as opposed to use of the  Authorization Code or the Client Credentials grants. Could this be related to the aouth issues that we have been  seeing?
